source/documentation/tools/create-a-derived-table/style-guide.md

```diff
@@ -0,0 +1,415 @@
+
+## 'Import' CTEs
+
+- 🔝 All `{{ ref('...') }}` statements should be placed in CTEs at the top of the file.
+- 📦 'Import' CTEs should be named after the table they are referencing.
+- 'Import' CTEs should be short and concise to make it easy to read what tables are being read in. Try not to have lots of fields in your select statement as it makes reading the CTEs harder, `select *` should be sufficient in most cases.
+- You can use a `where` clause to filter out any data you don't need.
+- For example:
+
+```sql
+with
+
+requirements as (
+    select * from {{ ref("derived_delius_stg__base_rqmnt") }}
+),
+
+reference_1 as (
+    select * from {{ ref("derived_delius_stg__stg_standard_reference_1") }} 
+    where code_set_name = 'REQUIREMENT SUB CATEGORY'
+),
+
+reference_2 as (
+    select * from {{ ref("derived_delius_stg__stg_standard_reference_2") }}
+    where code_set_name = 'ADDITIONAL REQUIREMENT SUB CATEGORY'
+),
+
+reference_3 as (
+    select * from {{ ref("derived_delius_stg__stg_standard_reference_3") }}
+    where code_set_name = 'REQUIREMENT TERMINATION REASON'
+),
+
+reference_4 as (
+    select * from {{ ref("derived_delius_stg__stg_standard_reference_4") }}
+    where code_set_name = 'UNITS'  
+),
+
+joined as (
+
+    select
+    requirements.rqmnt_id,
+    requirements.disposal_id,
+    requirements.start_date,
+    requirements.length,
+    requirements.rqmnt_notes,
+    requirements.commencement_date,
+    requirements.termination_date,
+    requirements.partition_area_id,
+    requirements.expected_start_date,
+    requirements.soft_deleted,
+    requirements.expected_end_date,
+
+    -- For rqmnt main type sub-cat
+    reference_1.code_value as rqmnt_type_sub_category_code,
+    reference_1.code_description as rqmnt_type_sub_category_desc,
+
+    -- For rqmnt main type sub-cat
+    reference_2.code_value as ad_rqmnt_type_sub_category_code,
+    reference_2.code_description as ad_rqmnt_type_sub_category_desc,
+
+    -- For terminations
+    reference_3.code_value as rqmnt_termination_reason_code,
+    reference_3.code_description as rqmnt_termination_reason_desc,
+
+    -- For units, bring code_value and code_description from standard_reference_list_id
+    reference_4.code_value as rqmnt_type_main_units_code,
+    reference_4.code_description as rqmnt_type_main_units_desc
+
+    from requirements
+
+    left join reference_1
+        on requirements.rqmnt_type_sub_category_id = reference_1.standard_reference_list_id
+
+    left join reference_2
+        on requirements.ad_rqmnt_type_sub_category_id = reference_2.standard_reference_list_id
+
+    left join reference_3
+        on requirements.rqmnt_termination_reason_id = reference_3.standard_reference_list_id
+
+    left join reference_4
+        on requirements.units_id = reference_4.standard_reference_list_id
+)
+
+select * from joined
+```
+
+## 'Functional' CTEs
```

Comment:
worth here stating what CTEs are. some users may use them without knowing what they are

Reply:
I'll add a link to a definition https://docs.getdbt.com/terms/cte